src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx

```diff
@@ -132,18 +133,37 @@ export const useTradeExecution = (
       dispatch(tradeQuoteSlice.actions.setSwapTxPending({ hopIndex, id: confirmedTradeId }))
 
       const onFail = (e: unknown) => {
-        const message = (() => {
+        const { message, translationKey, error } = (() => {
           if (e instanceof ChainAdapterError) {
```

Comment:
Another suggestion which would help achieving what we want to here (keep error-handling colocated with its swapper): solana's `executeSolanaTransaction` is currently just consuming the base implementation of `executeSolanaTransaction` from `swapper/src/utils`.

If you wrap `executeSolanaTx` into an handler, then you can actually do proper error-handling there and return something better
```diff
diff --git a/packages/swapper/src/swappers/JupiterSwapper/JupiterSwapper.ts b/packages/swapper/src/swappers/JupiterSwapper/JupiterSwapper.ts
index eb1966aead..deeba7c500 100644
--- a/packages/swapper/src/swappers/JupiterSwapper/JupiterSwapper.ts
+++ b/packages/swapper/src/swappers/JupiterSwapper/JupiterSwapper.ts
@@ -1,0 +2 @@ import type { AssetId } from '@shapeshiftoss/caip'
+import { ChainAdapterError } from '@shapeshiftoss/chain-adapters'
@@ -7,0 +9,7 @@ import { jupiterSupportedChainIds } from './utils/constants'
+export class SolanaLogsError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'SolanaLogsError'
+  }
+}
+
@@ -9 +17,16 @@ export const jupiterSwapper: Swapper = {
-  executeSolanaTransaction,
+  executeSolanaTransaction: async (...args) => {
+    try {
+      const txid = await executeSolanaTransaction(...args)
+      return txid
+    } catch (e: unknown) {
+      if (e instanceof ChainAdapterError) {
+        // replace me with actual pattern matching
+        const isLogsError = true
+        // where errorMember is the actual error from enum
+        if (isLogsError) throw new SolanaLogsError(SLIPPAGE_EXCEEDED)
+      }
+
+      // rethrow is not instanceof ChainAdapterError, shouldn't be the case but jus to satisfy TS
+      throw e
+    }
+  },
diff --git a/src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx b/src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx
index c87f2e4589..ea4894c0e1 100644
--- a/src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx
+++ b/src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx
@@ -24,0 +25 @@ import { LIFI_TRADE_POLL_INTERVAL_MILLISECONDS } from '@shapeshiftoss/swapper/di
+import { SolanaLogsError } from '@shapeshiftoss/swapper/src/swappers/JupiterSwapper/JupiterSwapper'
@@ -136,0 +138,3 @@ export const useTradeExecution = (
+          if (e instanceof SolanaLogsError) {
+            // switch error enum member to translation here
+          }
```

Reply:
Implemented in [17f168a](https://github.com/shapeshift/web/pull/8433/commits/17f168a1ba52efd3e9b3f5e55e1b8f1e2266db37)